### PR TITLE
Mini Apps: Add section on Mini App Updates

### PR DESCRIPTION
**Pull request by cmfcmf**
This adds a small section to the Mini App article series about how updates work for Mini Apps.

### CODE REVIEW

**Comment by cmfcmf** [comment]
@tomayac please take a look :)

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 1]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+their mini apps. A super app then uses that backend system to check for and download updates. Some super apps
+perform updates entirely in the background, without any way for the mini app itself to influence the update
+flow. Other super apps give more control to the mini apps themselves.
+
+The following paragraphs describe [WeChat's update mechanism for mini apps](https://developers.weixin.qq.com/miniprogram/dev/framework/runtime/update-mechanism.html)
+in more detail. WeChat checks for available updates in the following two scenarios:
+
+1. WeChat will regularly check for updates of recently used mini apps as long as WeChat is running. If an
+   update is found, the update is downloaded, and synchronously applied the next time the user cold-starts the
+   mini app. "Cold starting" a mini apps occurs when the mini app was not currently running when the user
```
Comment (tomayac) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:287]:
```suggestion
   mini app. Cold-starting a mini apps occurs when the mini app was not currently running when the user
```

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 2]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+   opened it (WeChat force-closes mini apps after being in the background for 5 minutes).
```
Comment (tomayac) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:288]:
```suggestion
   opened it (WeChat force-closes mini apps after being in the background for 5&nbsp;minutes).
```

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 3]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+2. WeChat also checks for updates when a mini app is cold-started. For mini apps that the user has not opened
+   "for a long time", the update is checked for and downloaded synchronously. While the update is downloading,
```
Comment (tomayac) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:290]:
```suggestion
   for a long time, the update is checked for and downloaded synchronously. While the update is downloading,
```

You don't use air quotes further down, where you say "recently". I think it's fine to remain vague at both locations.

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 4]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+- notifies the mini app when an update has been downloaded and is available
+  ([`onUpdateReady`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.onUpdateReady.html))
+- notifies the mini app when an update could not be downloaded
+  ([`onUpdateFailed`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.onUpdateFailed.html))
+- allows the mini app to force-install an available update, which will restart the app
+  ([`applyUpdate`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.applyUpdate.html))
+
+WeChat also provides additional update customization options for mini app developers in its backend system:
+1. One option allows developers to opt-out from synchronous updates for users that already have a certain
+   minimum version of the mini app installed, and instead force the updates to be asynchronous.
```
Comment (tomayac) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:309]:
```suggestion
   minimum version of the mini app installed, and instead forces the updates to be asynchronous.
```

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 5]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
 
+## Updates
+
+Mini apps do not have a standardized update mechanism ([discussion around potential standardization](https://github.com/w3c/miniapp-packaging/issues/28)).
+All mini app platforms have a backend system, which allows mini app developers to upload new versions of
+their mini apps. A super app then uses that backend system to check for and download updates. Some super apps
+perform updates entirely in the background, without any way for the mini app itself to influence the update
+flow. Other super apps give more control to the mini apps themselves.
+
+The following paragraphs describe [WeChat's update mechanism for mini apps](https://developers.weixin.qq.com/miniprogram/dev/framework/runtime/update-mechanism.html)
```
Comment (tomayac) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:282]:
```suggestion
As an example of a sophisticated process, the following paragraphs describe [WeChat's update mechanism for mini apps](https://developers.weixin.qq.com/miniprogram/dev/framework/runtime/update-mechanism.html)
```

**Review by tomayac — APPROVED** [review]
LGTM, thank you! Left a couple of suggestions.

CC: @anaclumos, you may be interested in adding this section in the corresponding chapter of the Korean translation!

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md (hunk 3 above)
Comment (cmfcmf) [reply src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md]: Fair!

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 6]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+2. WeChat also checks for updates when a mini app is cold-started. For mini apps that the user has not opened
+   for a long time, the update is checked for and downloaded synchronously. While the update is downloading,
+   the user has to wait. Once the download finishes, the update is applied, and the mini app opens. If the
+   download fails, e.g., due to bad network connectivity, the mini app opens regardless. For mini apps that
+   the user has opened recently, any potential update is downloaded asynchronously in the background and will
+   be applied the next time the user cold-starts the mini app.
+
+Mini apps can opt-in to earlier updates by using the [`UpdateManager`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.html) API. It provides the following functionality:
+
+- notifies the mini app when a check for updates is made
```
Comment (rachelandrew) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:298]:
```suggestion
- Notifies the mini app when a check for updates is made.
```

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 7]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+  ([`onCheckForUpdate`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.onCheckForUpdate.html))
+- notifies the mini app when an update has been downloaded and is available
```
Comment (rachelandrew) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:300]:
```suggestion
- Notifies the mini app when an update has been downloaded and is available.
```

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 8]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+  ([`onUpdateReady`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.onUpdateReady.html))
+- notifies the mini app when an update could not be downloaded
```
Comment (rachelandrew) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:302]:
```suggestion
- Notifies the mini app when an update could not be downloaded.
```

File: src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md [hunk 9]
```diff
@@ -270,6 +271,47 @@ tt.canIUse("request.success.data");
...
+  ([`onUpdateFailed`](https://developers.weixin.qq.com/miniprogram/en/dev/api/base/update/UpdateManager.onUpdateFailed.html))
+- allows the mini app to force-install an available update, which will restart the app
```
Comment (rachelandrew) [inline src/site/content/en/mini-apps/mini-app-markup-styling-and-scripting/index.md:304]:
```suggestion
- Allows the mini app to force-install an available update, which will restart the app.
```

**Review by rachelandrew — APPROVED** [review]
With a bit of punctuation to commit before merging.

**Comment by cmfcmf** [comment]
Thanks Rachel, done!

**Comment by anaclumos** [comment]
> LGTM, thank you! Left a couple of suggestions.
> 
> CC: @anaclumos, you may be interested in adding this section in the corresponding chapter of the Korean translation!

Sure! I'll do that once this draft gets published.

**Comment by tomayac** [comment]
This is live on the site: https://web.dev/articles/mini-apps/mini-app-markup-styling-and-scripting#updates. Closing the PR as it was added on the new infrastructure via a change list that reflects this PR. Thanks for adding this section!

**Pull request closed without merging**